### PR TITLE
#10257: Add ttnn binding for UnaryWithParam, UnaryOpType

### LINKED ISSUE #10257: Add UnaryWithParam, UnaryOpType binding to TTNN and replace FusibleActivation
- [x] Create new binding for FusibleActivation in TTNN library to replace reference to tt_eager  and eltwise_unary_op.hpp
- [x] Change the reference used in `ttnn.UnaryWithParam` and `ttnn.UnaryOpType` to use the new binding 
```
UnaryWithParam = ttnn._ttnn.deprecated.tensor.FusibleActivationWithParam
UnaryOpType = ttnn._ttnn.deprecated.tensor.FusibleActivation
```

- [x] To replace references of the function UnaryWithParam and enum UnaryOpType to ttnn/eltwise/unary

```
using ttnn::operations::unary::UnaryWithParam;
using ttnn::operations::unary::UnaryOpType;
```

- [x] UnaryWithParam is used as a parameter `std::optional<UnaryWithParam> `in binary ops, matmul ops etc. 

- [x] Replace usage of tt_lib.tensor.FusibleActivation in models and tests

### PR DESCRIPTION
**Pull request by KalaivaniMCW**
### Ticket
Link to Github Issue #10257 

### Problem description
ttnn.UnaryWithParam, ttnn.UnaryOpType refers to `tt_lib.tensor.FusibleActivationWithParam` and `tt_lib.tensor.FusibleActivation`
Add new UnaryWithParam, UnaryOpType binding to TTNN and replace the tt_eager references

### What's changed
Created binding for UnaryWithParam function and enum UnaryOpType from ttnn/eltwise/unary

- [x] Add new UnaryWithParam, UnaryOpType binding to TTNN 
- [x] Changed reference in binary, matmul, loss, conv, optimizer ops
- [x] Changed references in cpp files
- [x] Changed references in py files - tests and models

### Checklist
- [x] Post commit CI passes - https://github.com/tenstorrent/tt-metal/actions/runs/9979858229
- [ ] Nightly fast dispatch  - https://github.com/tenstorrent/tt-metal/actions/runs/9979856672
- [x] [post-commit] models tests - https://github.com/tenstorrent/tt-metal/actions/runs/9972762931
- [x] Device perf regressions and output report -  https://github.com/tenstorrent/tt-metal/actions/runs/9974449735
- [ ] Model perf regressions and output report - https://github.com/tenstorrent/tt-metal/actions/runs/9974447217
- [x] T3K unit tests - https://github.com/tenstorrent/tt-metal/actions/runs/9974444254
- [x] T3K frequent tests - https://github.com/tenstorrent/tt-metal/actions/runs/9974457659
- [x] T3K demo tests - https://github.com/tenstorrent/tt-metal/actions/runs/9971393558
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: ttnn/ttnn/__init__.py [hunk 1]
```diff
@@ -25,8 +25,11 @@
 
 CPP_CONFIG: ttnn._ttnn.core.Config = ttnn._ttnn.CONFIG
 
-UnaryWithParam = ttnn._ttnn.deprecated.tensor.FusibleActivationWithParam
-UnaryOpType = ttnn._ttnn.deprecated.tensor.FusibleActivation
+# UnaryWithParam = ttnn._ttnn.deprecated.tensor.FusibleActivationWithParam
+# UnaryOpType = ttnn._ttnn.deprecated.tensor.FusibleActivation
+
+UnaryWithParam = ttnn._ttnn.activation.ActivationUnaryWithParam
+UnaryOpType = ttnn._ttnn.activation.ActivationUnaryOpType
```
Comment (eyonland) [inline ttnn/ttnn/__init__.py:32]: We want parity between the Python and C++.   For example ttnn.activation._ttnn.UnaryOpType

File: ttnn/cpp/pybind11/activation.hpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+namespace py = pybind11;
+
+namespace ttnn {
+namespace activation {
+void py_module(py::module& m) {
+    using namespace ttnn::operations::unary;
+
+    // Export the UnaryOpType enum using the export_enum utility
+    export_enum<UnaryOpType>(m, "ActivationUnaryOpType");
```
Comment (arakhmati) [inline ttnn/cpp/pybind11/activation.hpp:23]: please rename to `ActivationUnaryOpType ` from `UnaryOpType`

File: ttnn/cpp/pybind11/activation.hpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    // Export the UnaryWithParam class
+    py::class_<UnaryWithParam>(m, "ActivationUnaryWithParam")
```
Comment (arakhmati) [inline ttnn/cpp/pybind11/activation.hpp:26]: please rename to `ActivationUnaryWithParam ` from `UnaryWithParam`

**Review by arakhmati — APPROVED** [review]

File: ttnn/cpp/pybind11/activation.hpp (hunk 2 above)
Comment (arakhmati) [reply ttnn/cpp/pybind11/activation.hpp:23]: I will approve once both are renamed

**Review by arakhmati — CHANGES_REQUESTED** [review]

File: ttnn/cpp/pybind11/activation.hpp (hunk 3 above)
Comment (KalaivaniMCW) [reply ttnn/cpp/pybind11/activation.hpp:26]: I'll rename the class to "UnaryWithParam" same as the struct in ttnn/unary it refers to. is that ok ?

Comment (KalaivaniMCW) [reply ttnn/cpp/pybind11/activation.hpp]: image

File: ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp [hunk 4]
```diff
@@ -642,7 +643,7 @@ inline MatmulProgramConfig generate_matmul_program_config(
     const MemoryConfig& mem_config,
     const std::optional<const DeviceComputeKernelConfig> compute_kernel_config,
     const std::optional<const CoreCoord> user_core_coord,
-    const std::optional<const UnaryWithParam> user_fused_activation,
+    const std::optional<UnaryWithParam> user_fused_activation,
```
Comment (bbradelTT) [inline ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp:646]: Why remove the const?

File: ttnn/cpp/ttnn/operations/matmul/device/matmul_op.hpp [hunk 5]
```diff
@@ -7,7 +7,8 @@
 
 #include "tensor/tensor.hpp"
 
-#include "tt_dnn/op_library/eltwise_unary/eltwise_unary_op.hpp"
+// #include "tt_dnn/op_library/eltwise_unary/eltwise_unary_op.hpp"
```
Comment (bbradelTT) [inline ttnn/cpp/ttnn/operations/matmul/device/matmul_op.hpp:10]: Why is the include commented out instead of deleted?

Comment (KalaivaniMCW) [reply ttnn/cpp/ttnn/operations/matmul/device/matmul_op.hpp:10]: @bbradelTT will safely remove these when all CI pipelines are green.

File: ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp (hunk 4 above)
Comment (KalaivaniMCW) [reply ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp:646]: @bbradelTT In Matmul, both `std::optional<UnaryWithParam>` and ` const std::optional<const UnaryWithParam>` are used at different places. and using <const UnaryWithParam> throws an error.

File: ttnn/cpp/ttnn/operations/matmul/device/matmul_op.hpp (hunk 5 above)
Comment (bbradelTT) [reply ttnn/cpp/ttnn/operations/matmul/device/matmul_op.hpp:10]: Ok. Thanks.

File: ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp (hunk 4 above)
Comment (bbradelTT) [reply ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp:646]: I think they should all be const. I'll need to clean that up afterwards.

Comment (bbradelTT) [reply ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp:646]: @KalaivaniMCW are you able to leave this as non-const for now? If not, what issues are you running into for the matmul code (@eyonland mentioned that you were running into some issues)?

Comment (KalaivaniMCW) [reply ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp:646]: @bbradelTT I am able to use std::optional<UnaryWithParam> without any issues.

**Review by boris-drazic — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/eltwise/unary/alias.hpp [hunk 6]
```diff
@@ -0,0 +1,12 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
```
Comment (ayerofieiev-tt) [inline ttnn/cpp/ttnn/operations/eltwise/unary/alias.hpp:1]: This file is not used anywhere as far as I see. Lets remove?

**Review by ayerofieiev-tt — APPROVED** [review]

**Review by bbradelTT — APPROVED** [review]

**Review by skhorasganiTT — APPROVED** [review]

**Review by arakhmati — APPROVED** [review]

**Review by cglagovichTT — APPROVED** [review]

**Review by eyonland — APPROVED** [review]

**Review by mywoodstock — APPROVED** [review]

**Pull request merged**